packages/dds/shared-object-base/src/remoteObjectHandle.ts

```diff
@@ -43,7 +43,8 @@ export class RemoteFluidObjectHandle implements IFluidHandle {
 
     public async get(): Promise<any> {
         if (this.objectP === undefined) {
-            const request = { url: this.absolutePath };
+            // Add `viaHandle` header to distinguish from requests from non-handle paths.
+            const request: IRequest = { url: this.absolutePath, headers: { viaHandle: true } };
```

Comment:
I'd suggest having enum to represent all resolution modes, vs. a bunch of boolean flags in various places.

Reply:
Added enums for this.